cmd/packaging/packaging.go

```diff
@@ -162,15 +157,166 @@ func runDockerPackaging(path string, packagingFormat string, command []string) {
 	}
 }
 
+var templateData map[string]string
+var once sync.Once
+
 func getTemplateData(projectName, buildVersion string) map[string]string {
-	return map[string]string{
-		"projectName":         projectName,
-		"strippedProjectName": removeDashesAndUnderscores(projectName),
-		"author":              getAuthor(),
-		"version":             buildVersion,
-		"description":         pubspec.GetPubSpec().Description,
-		"debDependencies":     strings.Join(linuxPackagingDependencies, ","),
-		"snapDependencies":    strings.Join(linuxPackagingDependencies, "\n      - "),
-		"organizationName":    androidmanifest.AndroidOrganizationName(),
+	once.Do(func() {
+		templateData = map[string]string{
+			"projectName":         projectName,
+			"strippedProjectName": strings.ReplaceAll(projectName, "_", ""),
+			"author":              getAuthor(),
+			"version":             buildVersion,
+			"description":         pubspec.GetPubSpec().Description,
+			"organizationName":    androidmanifest.AndroidOrganizationName(),
+			"arch":                runtime.GOARCH,
+		}
+	})
+	return templateData
+}
+
+type packagingTask struct {
+	packagingFormatName            string
+	dependsOn                      map[*packagingTask]string
+	templateFiles                  map[string]string
+	executableFiles                []string
+	linuxDesktopFileExecutablePath string
+	linuxDesktopFileIconPath       string
+	dockerfileContent              []string
+	generateBuildFiles             func(projectName, path string)
+	buildOutputDirectory           string
+	packagingCommand               string
+	resultFileName                 string
+	outputFileExtension            string
+}
+
+func (t *packagingTask) Init(ignore ...bool) {
```

Comment:
I would really prefer to not use `...` for optional arguments because it's not descriptive. What happens if I write `t.Init(true, false, true, true)` ??.

Maybe create an exported method `Init()` that just calls the unexported method `init(ignore bool)` with `false` as argument.
That way, external use of the Init function is forced to ignore=false, and in this code (inside the package) it is always explicitly given (the calls to `Init(true)` below rewritten to `init(true)`)